### ISSUE TITLE
Cannot connect to Solus Budgie desktop via RDP

### ISSUE THREAD

**Issue by A108384** [getsolus/packages#1524]
### Summary

I have a recent Solus installation on a new computer with current updates.  It works great when logged in directly.   
My issue is that I am now trying to connect to it from another computer also running Solus.   I would specifically like to get xRDP working so that I can connect to it via Remmina.  

I have installed xrdp on the target Solus machine.  When I attempt to connect via RDP using Remmina I get the xrdp login gui window, where I enter my userid/password, then I get another pop-up that says:  "login failed for user <xxxx>".   

I see this in /var/log/xrdp-sesman.log:  
   [20240204-09:35:05] [INFO ] Socket 12: AF_INET6 connection received from ::1 port 59322
   [20240204-09:35:06] [ERROR] pam_authenticate failed: Authentication failure
   [20240204-09:35:06] [INFO ] AUTHFAIL: user=<xxxx> ip=::ffff:10.24.0.213 time=1707057306
   [20240204-09:35:06] [ERROR] sesman_data_in: scp_process_msg failed
   [20240204-09:35:06] [ERROR] sesman_main_loop: trans_check_wait_objs failed, removing trans

and this via journalctl:
   [20240204-09:35:05] [INFO ] Socket 12: AF_INET6 connection received from ::1 port 59322
   [20240204-09:35:06] [ERROR] pam_authenticate failed: Authentication failure
   [20240204-09:35:06] [INFO ] AUTHFAIL: user=<xxxx> ip=::ffff:10.24.0.213 time=1707057306
   [20240204-09:35:06] [ERROR] sesman_data_in: scp_process_msg failed
   [20240204-09:35:06] [ERROR] sesman_main_loop: trans_check_wait_objs failed, removing trans

Eventually I tried logging in as root and that is successful.  So root works.  I can use RDP to the remote machine when using root credentials.   But it does not work when using my personal credentials

I would appreciate any additional assistance.     

### Steps to reproduce

1.  Install xrdp and xrdp-budgie-integration on the target machine.  (I also installed other things - like vnc - while trying to get rdp to work.  I don't have a list of everything I installed trying to resolve the issue)
2. Install Remmina on the local machine.
3. In Remmina - Define a new RDP connection to the target Solus machine.  (I have tried lots of different Remmina settings but none made a difference for me.   So you can just define a basic RDP connection with no special settings.  Just the target hostname)
4. Save the Connection
5. Double-click on the new connect to connect to the target machine.  
6. I get a gui screen with a xrdp prompt for my userid and password.    (Xorg is listed as the Session)
   
![image](https://github.com/getsolus/packages/assets/153583686/4633f828-fa35-42d9-9cf3-173eb74cf284)

7. Enter userid and password.  Then I get a new pop-up indicating that my login failed.
![image](https://github.com/getsolus/packages/assets/153583686/945669b3-2a56-439c-bfc6-b2c69ef260cf)

However, if I enter the root user and password on the xrdp authentication screen - that successfully works.  A Solus Budgie desktop is launched and I am in as root.   
I have not been able to determine why this does not work for my normal user account.

### Expected result

I expect to be presented with a Budgie Desktop.

### Actual result

I get a pop-up that my login failed.

### Environment

- [X] Is system up to date?

### Repo

Shannon (stable)

### Desktop Environment

Budgie

### System details

inxi -b  output from the target system:
    Last login: Sun Feb  4 09:28:21 2024 from 10.24.0.213
    cvg01@kvm-server ~ $ inxi -b
    System:
      Host: kvm-server Kernel: 6.6.15-272.current arch: x86_64 bits: 64 Console: pty pts/0
        Distro: Solus 4.5 resilience
    Machine:
      Type: Mini-pc System: Intel Client Systems product: NUC9i5QNX v: K49246-404
        serial: <superuser required>
      Mobo: Intel model: NUC9i5QNB v: K49247-403 serial: <superuser required> UEFI: Intel
        v: QXCFL579.0045.2020.0617.1710 date: 06/17/2020
    CPU:
      Info: quad core Intel Core i5-9300H [MT MCP] speed (MHz): avg: 800 min/max: 800/4100
    Graphics:
      Device-1: Intel CoffeeLake-H GT2 [UHD Graphics 630] driver: i915 v: kernel
      Display: server: X.org v: 1.21.1.11 with: Xwayland v: 23.2.4 driver: X:
        loaded: fbdev,modesetting,vesa gpu: i915 tty: 175x37
      API: OpenGL v: 4.6 compat-v: 4.5 vendor: mesa v: 23.3.5 note: console (EGL sourced)
        renderer: Mesa Intel UHD Graphics 630 (CFL GT2), llvmpipe (LLVM 16.0.6 256 bits)
    Network:
      Device-1: Intel Ethernet I219-LM driver: e1000e
      Device-2: Intel Wi-Fi 6 AX200 driver: iwlwifi
      Device-3: Intel I210 Gigabit Network driver: igb
    Drives:
      Local Storage: total: 2.26 TiB used: 808.14 GiB (35.0%)
    Info:
      Processes: 241 Uptime: 2h 54m Memory: total: 32 GiB note: est. available: 31.07 GiB
      used: 848.4 MiB (2.7%) Init: systemd target: graphical (5) Shell: Bash inxi: 3.3.31

inxi -b output from the source system:
    System:
      Host: commander Kernel: 6.6.15-272.current arch: x86_64 bits: 64
        Desktop: Budgie v: 10.9.1 Distro: Solus 4.5 resilience
    Machine:
      Type: Laptop System: LENOVO product: 20KHCTO1WW v: ThinkPad X1 Carbon 6th
        serial: <superuser required>
      Mobo: LENOVO model: 20KHCTO1WW v: SDK0J40697 WIN
        serial: <superuser required> UEFI: LENOVO v: N23ET55W (1.30 )
        date: 08/31/2018
    Battery:
      ID-1: BAT0 charge: 49.3 Wh (99.4%) condition: 49.6/57.0 Wh (87.0%)
    CPU:
      Info: quad core Intel Core i7-8550U [MT MCP] speed (MHz): avg: 515
        min/max: 400/4000
    Graphics:
      Device-1: Intel UHD Graphics 620 driver: i915 v: kernel
      Device-2: Logitech HD Pro Webcam C920 driver: snd-usb-audio,uvcvideo
        type: USB
      Device-3: IMC Networks Integrated Camera driver: uvcvideo type: USB
      Display: x11 server: X.Org v: 21.1.11 with: Xwayland v: 23.2.4 driver: X:
        loaded: modesetting unloaded: fbdev,vesa dri: iris gpu: i915 resolution:
        1: 3840x1200~85Hz 2: N/A
      API: OpenGL v: 4.6 compat-v: 4.5 vendor: intel mesa v: 23.3.5
        renderer: Mesa Intel UHD Graphics 620 (KBL GT2)
    Network:
      Device-1: Intel Ethernet I219-V driver: e1000e
      Device-2: Intel Wireless 8265 / 8275 driver: iwlwifi
      Device-3: Lenovo ThinkPad Dock Ethernet [Realtek RTL8153B] driver: r8152
        type: USB
    Drives:
      Local Storage: total: 6.37 TiB used: 1013.83 GiB (15.5%)
    Info:
      Processes: 344 Uptime: 1d 2h 23m Memory: total: 16 GiB note: est.
      available: 15.4 GiB used: 6.43 GiB (41.7%) Shell: Bash inxi: 3.3.31

### Other comments

Journalctl output for user login (cvg01)
Feb 04 12:24:59 kvm-server xrdp[2701]: [INFO ] connecting to sesman on 127.0.0.1:3350
Feb 04 12:24:59 kvm-server xrdp-sesman[1758]: [INFO ] Socket 12: AF_INET6 connection received from ::1 port 56836
Feb 04 12:24:59 kvm-server xrdp[2701]: [INFO ] xrdp_wm_log_msg: sesman connect ok
Feb 04 12:24:59 kvm-server xrdp[2701]: [INFO ] sesman connect ok
Feb 04 12:24:59 kvm-server xrdp[2701]: [INFO ] sending login info to session manager. Please wait...
Feb 04 12:24:59 kvm-server xrdp-sesman[1758]: pam_unix(xrdp-sesman:auth): authentication failure; logname= uid=0 euid=0 tty=xrdp-sesman ruser= rhost=  user=cvg01
Feb 04 12:25:01 kvm-server xrdp-sesman[1758]: [ERROR] pam_authenticate failed: Authentication failure
Feb 04 12:25:01 kvm-server xrdp-sesman[1758]: [INFO ] AUTHFAIL: user=cvg01 ip=::ffff:10.24.0.213 time=1707067501
Feb 04 12:25:01 kvm-server xrdp-sesman[1758]: [ERROR] sesman_data_in: scp_process_msg failed
Feb 04 12:25:01 kvm-server xrdp[2701]: [INFO ] xrdp_wm_log_msg: login failed for user cvg01
Feb 04 12:25:01 kvm-server xrdp[2701]: [INFO ] login failed for user cvg01
Feb 04 12:25:01 kvm-server xrdp-sesman[1758]: [ERROR] sesman_main_loop: trans_check_wait_objs failed, removing trans

journalctl output for root login (works):
Feb 04 12:26:55 kvm-server xrdp[2701]: [INFO ] connecting to sesman on 127.0.0.1:3350
Feb 04 12:26:55 kvm-server xrdp-sesman[1758]: [INFO ] Socket 12: AF_INET6 connection received from ::1 port 37538
Feb 04 12:26:55 kvm-server xrdp[2701]: [INFO ] xrdp_wm_log_msg: sesman connect ok
Feb 04 12:26:55 kvm-server xrdp[2701]: [INFO ] sesman connect ok
Feb 04 12:26:55 kvm-server xrdp[2701]: [INFO ] sending login info to session manager. Please wait...
Feb 04 12:26:55 kvm-server xrdp-sesman[1758]: gkr-pam: no password is available for user
Feb 04 12:26:55 kvm-server xrdp-sesman[1758]: [INFO ] Terminal Server Users group is disabled, allowing authentication
Feb 04 12:26:55 kvm-server xrdp-sesman[1758]: [INFO ] ++ created session (access granted): username root, ip ::ffff:10.24.0.213:56036 - socket: 12
Feb 04 12:26:55 kvm-server xrdp-sesman[1758]: [INFO ] starting Xorg session...
Feb 04 12:26:55 kvm-server xrdp-sesman[1758]: [INFO ] Starting session: session_pid 2706, display :10.0, width 1388, height 896, bpp 24, client ip ::ffff:10.24.0.213:56036 - >
Feb 04 12:26:55 kvm-server xrdp-sesman[2706]: [INFO ] [session start] (display 10): calling auth_start_session from pid 2706
Feb 04 12:26:55 kvm-server xrdp-sesman[1758]: [ERROR] sesman_data_in: scp_process_msg failed
Feb 04 12:26:55 kvm-server xrdp[2701]: [INFO ] xrdp_wm_log_msg: login successful for user root on display 10
Feb 04 12:26:55 kvm-server xrdp[2701]: [INFO ] login successful for user root on display 10
Feb 04 12:26:55 kvm-server xrdp-sesman[1758]: [ERROR] sesman_main_loop: trans_check_wait_objs failed, removing trans
Feb 04 12:26:55 kvm-server xrdp[2701]: [INFO ] loaded module 'libxup.so' ok, interface size 10296, version 4
Feb 04 12:26:55 kvm-server xrdp-sesman[2706]: pam_unix(xrdp-sesman:session): session opened for user root(uid=0) by root(uid=0)
Feb 04 12:26:55 kvm-server xrdp[2701]: [INFO ] started connecting
Feb 04 12:26:55 kvm-server xrdp[2701]: [INFO ] lib_mod_connect: connecting via UNIX socket
Feb 04 12:26:55 kvm-server systemd-logind[755]: New session 10 of user root.
Feb 04 12:26:55 kvm-server systemd[1]: Created slice User Slice of UID 0.

**Comment by A108384** [comment]
I finally figured out the issue.   (long story, but i'll be brief.  If you want more specific info, let me know)

The issue was with my keyboard mapping.  For some reason when i was using the shift key while typing my target host password it was not being interpreted properly.  So I was never actually entering a valid password for my account.

Back story.. I noticed a message in the xrdp log about the keyboard mapping it it not finding the proper km-00020409.ini file so xrdp was using a default file.   I still couldn't figure out what was really going on, but it put me on a search in that direction.  
I tried lots of things, and it was making a difference (breaking things) so at least I was on the right track maybe.  Anyway I finally found a keyboard layout setting in Remmina preferences.  There is a checkbox that says "Use client keyboard mapping".   When I unchecked this and connected i was able to pass authentication and was able to login in and get the remote desktop.   I guess originally my password (keyboard entry) was not be actually being interpreted properly.  But after making that change it could be for some reason.   
I assume that this option must be a Remmina default, but i'm not 100% sure.    And it probably works for most people.  Even I I have been using Remmina to connect to a windows host successfully for a very long time.  It just wouldn't work for this connection to another solus host.

Weird issue.  But it is now resolved.   So I was mark this closed.
